custom-metrics-stackdriver-adapter/adapter.go

```diff
@@ -211,6 +213,8 @@ func main() {
 		"Stackdriver Endpoint used by adapter. Default is https://monitoring.googleapis.com/")
 	flags.BoolVar(&serverOptions.EnableDistributionSupport, "enable-distribution-support", serverOptions.EnableDistributionSupport,
 		"enables support for scaling based on distribution values")
+	flags.DurationVar(&serverOptions.ExternalMetricCacheWindow, "external-metric-cache-window", serverOptions.ExternalMetricCacheWindow,
```

Comment:
for additional safety, should we add another boolean flag here to control the enablement of the cache?

Most users will want that off so we can completely bypass creating the cache, etc...

Reply:
So if externalMetricCacheWindow == 0, the cache will not be constructed and we effectively fall back to the previous flow. Does that suffice or do we still need an additional flag?

Reply:
I think that works!